universal-login-react/src/ui/Notifications/EmojiForm.tsx

```diff
@@ -4,45 +4,32 @@ import {
   isValidCode,
   Notification,
   SECURITY_CODE_LENGTH,
-  GasParameters,
+  ensureNotNull,
 } from '@universal-login/commons';
-import {DeployedWallet} from '@universal-login/sdk';
 import {EmojiPlaceholders} from './EmojiPlaceholders';
-import {transactionDetails} from '../../core/constants/TransactionDetails';
-import ProgressBar from '../commons/ProgressBar';
-import {useProgressBar} from '../hooks/useProgressBar';
-import {GasPrice} from '../commons/GasPrice';
-import {FooterSection} from '../commons/FooterSection';
 import {EmojiInput} from './EmojiInput';
 
 export interface EmojiFormProps {
-  deployedWallet: DeployedWallet;
-  onConnectionSuccess: () => void;
-  onDenyRequests?: () => void;
   hideTitle?: () => void;
   className?: string;
+  notifications: Notification[];
+  onCancelClick: () => void;
+  setPublicKey: (arg: string) => void;
 }
 
-export const EmojiForm = ({deployedWallet, hideTitle, className, onDenyRequests, onConnectionSuccess}: EmojiFormProps) => {
+export const EmojiForm = ({hideTitle, className, notifications, onCancelClick, setPublicKey}: EmojiFormProps) => {
   const [enteredCode, setEnteredCode] = useState<number[]>([]);
-  const {progressBar, showProgressBar} = useProgressBar();
-  const [gasParameters, setGasParameters] = useState<GasParameters | undefined>(undefined);
-
-  const [notifications, setNotifications] = useState<Notification[]>([]);
   const [soleAddress, setSoleAddress] = useState<string | undefined>(undefined);
 
-  useEffect(() => deployedWallet.subscribeAuthorisations(onNotificationsChange), []);
-
-  const onNotificationsChange = (notifications: Notification[]) => {
-    setNotifications(notifications);
-    updateSoleAddress(filterNotificationByCodePrefix(notifications, enteredCode));
-  };
+  useEffect(() => updateSoleAddress(filterNotificationByCodePrefix(notifications, enteredCode)), [notifications]);
```

Comment:
You can remove the call to `updateSoleAddress` in `updateEnteredCode` and just add `enteredCode` as a dependency to this effect.